pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -623,6 +623,10 @@ func validateNodeGroupBase(np NodePool, path string, controlPlaneOnOutposts bool
 	if instanceutils.IsInferentiaInstanceType(SelectInstanceType(np)) && ng.AMIFamily == NodeImageFamilyBottlerocket {
 		return errors.Errorf("Inferentia instance types are not supported for %s", ng.AMIFamily)
 	}
+	// Bottlerocket doesn't support Trainium hosts
+	if instanceutils.IsTrainiumInstanceType(SelectInstanceType(np)) && ng.AMIFamily == NodeImageFamilyBottlerocket {
```

Comment:
Windows AMIs do not ship with the drivers either and are not supported on Trainium. AFAIK, Ubuntu is not supported either, but it will still fail early because the AMI resolution will fail. 

This condition must be changed to allow only AL2 AMIs.

```suggestion
	if instanceutils.IsTrainiumInstanceType(SelectInstanceType(np)) && ng.AMIFamily != NodeImageFamilyAmazonLinux2 && ng.AMIFamily != "" {
```

This also applies to the Inferentia check above (which wasn't introduced in this PR).

Reply:
Nice catch.  I'll add it quick here.